src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -445,7 +453,7 @@ private function needsCdataWrapping(string $val): bool
      *
      * @throws NotEncodableValueException
      */
-    private function selectNodeType(\DOMNode $node, $val): bool
+    private function selectNodeType(\DOMNode $node, $val, $key = null): bool
```

Comment:
why make it an optional argument? i.e.: shouldn't the call inside appendNode also pass the key?
otherside of the question: if the anwer is not, what about dealing with the `#comment` check in buildXml directly, adding `} elseif ('#comment' === $key) { ...`?

Reply:
Right, dealing directly in `buildXml()` makes more sense because it's here that we manage the _special_ cases like `@`, `#` and now like `#comment`.